bucket/vcredist-aio.json

```
@@ -1,31 +1,31 @@
 {
-    "version": "0.102.0",
+    "version": "0.103.0",
     "description": "All-in-one repack for latest Microsoft Visual C++ Redistributable Runtimes, without the original setup bloat payload.",
-    "homepage": "https://github.com/abbodi1406/vcredist",
-    "license": "Unlicense",
-    "url": "https://github.com/abbodi1406/vcredist/releases/download/v0.102.0/VisualCppRedist_AIO_x86_x64_102.zip",
-    "hash": "53f7af87881747952697bad13f043ba2c303dcaefc8979ba7af140fab6db0ee8",
-    "installer": {
-        "script": [
-            "if (!(is_admin)) { error 'This package requires admin rights to install'; break }",
-            "Start-Process -Wait \"$dir\\VisualCppRedist_AIO_x86_x64.exe\" -ArgumentList @('/ai', '/gm2') -Verb RunAs"
-        ]
+    "homepage": "https://gitlab.com/stdout12/vcredist",
+    "license": {
+        "identifier": "Unlicense",
+        "url": "https://gitlab.com/stdout12/vcredist/-/blob/HEAD/LICENSE"
     },
-    "bin": "VisualCppRedist_AIO_x86_x64.exe",
-    "uninstaller": {
+    "notes": "View command-line options and usage examples by executing: `vcredist-aio.exe /?`",
+    "url": "https://gitlab.com/-/project/76069787/uploads/4431f3448fa76633cfd4983142f81952/VisualCppRedist_AIO_x86_x64.exe#/vcredist-aio.exe",
+    "hash": "3c188e4651bcc07df2bdb06d69585158797b1bafb91557b09217621628cd5b23",
+    "installer": {
         "script": [
-            "if (!(is_admin)) { error 'This package requires admin rights to uninstall'; break }",
-            "Start-Process -Wait \"$dir\\VisualCppRedist_AIO_x86_x64.exe\" -ArgumentList @('/ai', '/gm2') -Verb RunAs"
+            "if (-not (is_admin)) { error 'This package requires admin rights to install'; return }",
```

```suggestion
            "if (-not (is_admin)) { error 'This package requires admin rights to install'; break }",
```

Shouldn't the installation be aborted if the check fails?

---

>  they only require installation via `Start-Process` and do not need to support uninstallation

This is not true. I, for one, would indeed need to uninstall runtimes sometimes and given that vcredist-aio has auto uninstall mode, I don't think you should remove uninstaller script.

---

> Scoop should handle only downloading and performing basic guidance, while the subsequent management should be decided by the users themselves.

I totally agree. In my case, I only use `vcredist-aio` besides the ones preinstalled with Windows so when you say _universally applying full installation or full removal_, it's effectively the same thing as install/uninstall-ing `vcredist-aio` to me. I believe this makes it special enough to be the special case (i.e. keep uninstaller script) among other scoop runtime buckets.

Anyway, this is just personal and would not be a blocker. You can choose whichever path you like as I can always open `$SCOOP_GLOBAL\apps\vcredist-aio\currect` and uninstall it myself.

---

Hello, @Vinfall .

Thank you for your response, but I think my point may have been misunderstood.

The original uninstall script and install script were identical, both intended to silently install all packages without displaying progress. The correct uninstall script should use the `/aiR` parameter, but doing so would remove all detected runtimes, including those that were not installed by `vcredist-aio`. I consider this outcome counterproductive.

https://github.com/ScoopInstaller/Extras/blob/db8fae0940242a6c1a46354ea2ce873698270646/bucket/vcredist-aio.json#L15-L20

My intention is to remove the install and uninstall script sections entirely and instead use the `notes` field to provide the necessary explanations.

```powershell
┏[ ~]
└─> scoop cat Unofficial/vcredist-aio
─────┬──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
     │ STDIN
─────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1 │ {
   2 │     "version": "0.103.0",
   3 │     "description": "All-in-one repack for latest Microsoft Visual C++ Redistributable Runtimes, without the original setup bloat payload.",
   4 │     "homepage": "https://gitlab.com/stdout12/vcredist",
   5 │     "license": {
   6 │         "identifier": "Unlicense",
   7 │         "url": "https://gitlab.com/stdout12/vcredist/-/blob/HEAD/LICENSE"
   8 │     },
   9 │     "notes": [
  10 │         "Scoop is responsible only for downloading the app and does not install any runtimes.",
  11 │         "Install or manage the required runtimes by running the appropriate commands with administrative privileges.",
  12 │         "View command-line options and usage examples by executing: `vcredist-aio /?`",
  13 │         "For instance, to perform a silent installation of all packages without displaying progress, run:",
  14 │         "- `Start-Process -FilePath vcredist-aio -ArgumentList @('/ai', '/gm2') -Verb RunAs -Wait`"
  15 │     ],
  16 │     "url": "https://gitlab.com/-/project/76069787/uploads/4431f3448fa76633cfd4983142f81952/VisualCppRedist_AIO_x86_x64.exe#/vcredist-aio.exe",
  17 │     "hash": "3c188e4651bcc07df2bdb06d69585158797b1bafb91557b09217621628cd5b23",
  18 │     "bin": "vcredist-aio.exe",
  19 │     "checkver": {
  20 │         "url": "https://gitlab.com/api/v4/projects/76069787/releases/permalink/latest",
  21 │         "jsonpath": "$..['name', 'description']",
  22 │         "regex": "\"v([\\d.]+)\",\"[^\"]+\\[.+?_x86_x64\\.exe\\]\\(/uploads/(?<hash>\\w+)/"
  23 │     },
  24 │     "autoupdate": {
  25 │         "url": "https://gitlab.com/-/project/76069787/uploads/$matchHash/VisualCppRedist_AIO_x86_x64.exe#/vcredist-aio.exe",
  26 │         "hash": {
  27 │             "url": "https://gitlab.com/api/v4/projects/76069787/releases/permalink/latest",
  28 │             "regex": "$basename.+?SHA-256:\\s*$sha256"
  29 │         }
  30 │     }
  31 │ }
─────┴──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

┏[ ~]
└─> scoop install Unofficial/vcredist-aio
Installing 'vcredist-aio' (0.103.0) [64bit] from 'Unofficial' bucket
Loading VisualCppRedist_AIO_x86_x64.exe from cache.
Checking hash of VisualCppRedist_AIO_x86_x64.exe ... ok.
Linking D:\Software\Scoop\Local\apps\vcredist-aio\current => D:\Software\Scoop\Local\apps\vcredist-aio\0.103.0
Creating shim for 'vcredist-aio'.
Making D:\Software\Scoop\Local\shims\vcredist-aio.exe a GUI binary.
'vcredist-aio' (0.103.0) was installed successfully!
Notes
-----
Scoop is responsible only for downloading the app and does not install any runtimes.
Install or manage the required runtimes by running the appropriate commands with administrative privileges.
View command-line options and usage examples by executing: `vcredist-aio /?`
For instance, to perform a silent installation of all packages without displaying progress, run:
- `Start-Process -FilePath vcredist-aio -ArgumentList @('/ai', '/gm2') -Verb RunAs -Wait`
```

Even without an uninstall script, `vcredist-aio` can still be removed successfully. However, if you want to uninstall the runtimes installed through `vcredist-aio`, you must run `Start-Process -FilePath vcredist-aio -ArgumentList @('/aiR', '/gm2') -Verb RunAs -Wait` before executing `scoop uninstall vcredist-aio`.